Continue applying the new class naming convention

victornoel opened the pull request:
For #1003.

Since there was nothing to be renamed in set, I did the scalar package as well as some other that were lying around. So removed their `@todo` when relevant ;)

Also I left `UncheckedScalar` as a deprecated implementation based on `Unchecked` because cactoos-matchers relies on it. I added a `@todo` to resolve this situation later.

@llorllale I wonder if the list in https://github.com/yegor256/cactoos/issues/913#issuecomment-402332247 shouldn't be updated because it looks like new classes were introduced that could benefit from this new naming.

0crat commented:
Job [#1081](https://github.com/yegor256/cactoos/issues/1081) is now in scope, role is `REV`

<!-- https://www.0crat.com/footprint/C63314D6Z/ccfd547a-0c16-4482-864c-0c7c7eba1311, version: 0.43.8, hash: ${buildNumber} -->

0crat commented:
This pull request [#1081](https://github.com/yegor256/cactoos/issues/1081) is assigned to @iprogrammerr[/z](https://www.0crat.com/u/iprogrammerr), here is [why](https://www.0crat.com/footprint/C63314D6Z/418adaaa-6cb9-4294-94f8-4868a636d087); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

<!-- https://www.0crat.com/footprint/C63314D6Z/a19c24c8-ba1f-4410-805e-601c21d90b76, version: 0.43.8, hash: ${buildNumber} -->

Iprogrammerr commented on src/test/java/org/cactoos/scalar/IoCheckedTest.java:
```diff
@@ -29,18 +29,18 @@
  * @checkstyle JavadocMethodCheck (500 lines)
  */
-public final class IoCheckedScalarTest {
+public final class IoCheckedTest {
```
Could we use Throws in these tests?

victornoel replied:
@Iprogrammerr we could, but I spent enough time on this already :)

Iprogrammerr commented:
@victornoel Build failed. Besides that, I left you a review and add puzzle to continue, please:)

victornoel commented:
@Iprogrammerr thanks, I fixed the build, I updated the description to explain the problem with a circular dependency to cactoos-matchers and a class I had renamed.

victornoel commented:
@Iprogrammerr as for the puzzle to continue, I asked the question to @llorllale in the PR description, because I think all of the original renaming instructions are handled (except for text but it has its own puzzle).

llorllale commented:
@victornoel 

> Also I left UncheckedScalar as a deprecated implementation based on Unchecked because cactoos-matchers relies on it. I added a @todo to resolve this situation later.

Interesting. I remember you opened a ticket warning me about this a while ago.

>  I wonder if the list in #913 (comment) shouldn't be updated because it looks like new classes were introduced that could benefit from this new naming.

Which ones? I think we have more scalars to rename.

victornoel commented:
> Which ones? I think we have more scalars to rename.

In this PR I think all the scalar are done (I removed the `@todo` btw, luckily it was just assigned to me: #1025 :). There are some more Func, Input, Output that could maybe be renamed, not sure which one.

I rebased onto master, it should be alright now.

Iprogrammerr commented:
@victornoel Make build passing and we are good to go:)

llorllale commented:
@victornoel 

> There are some more Func, Input, Output that could maybe be renamed, not sure which one.

We missed `TimedFunc` and `FuncWithFallback`. I'm struggling to find a good name for the latter.

victornoel commented:
@llorllale I don't know why the build failed, I think it is a transient problem, any way to re-run it?

> We missed TimedFunc and FuncWithFallback. I'm struggling to find a good name for the latter.

I think at that time, there was also `TimedProc`, so you must have decided not to rename them (like the other `XXXFunc` and `XXXProc`).

Maybe it is not needed to rename `FuncWithFallback`?

victornoel commented:
@llorllale @Iprogrammerr I have force pushed to re-trigger CI, it works now

llorllale commented:
@victornoel 

> I think at that time, there was also TimedProc, so you must have decided not to rename them 

Right, that must have been it. Can you leave a `@todo` or open a ticket?

victornoel commented:
@llorllale well actually, I don't remember doing it but I already renamed `TimedFunc` to `Timed` :) so I think it's good now

llorllale commented:
@rultor merge

rultor commented:
> @rultor merge

@llorllale OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/18384-475868732)

0pdd commented:
@victornoel the puzzle [#1086](https://github.com/yegor256/cactoos/issues/1086) is still not solved.

rultor commented:
> @rultor merge

@llorllale Done! FYI, the full log is [here](http://www.rultor.com/t/18384-475868732) (took me 12min)

0crat commented:
Job `gh:yegor256/cactoos#1081` is not assigned, can't get performer

<!-- https://www.0crat.com/footprint/C63314D6Z/0119dce6-0623-4813-8441-fa9d05676915, version: 0.45.8, hash: ${buildNumber} -->

0crat commented:
The job [#1081](https://github.com/yegor256/cactoos/issues/1081) is now out of scope

<!-- https://www.0crat.com/footprint/C63314D6Z/d79db59c-7657-45c4-9714-d5dda8601fd0, version: 0.45.8, hash: ${buildNumber} -->

0pdd commented:
@victornoel the puzzle [#1144](https://github.com/yegor256/cactoos/issues/1144) is still not solved; solved: [#1086](https://github.com/yegor256/cactoos/issues/1086).

0pdd commented:
@victornoel all 2 puzzles are solved here: [#1086](https://github.com/yegor256/cactoos/issues/1086), [#1144](https://github.com/yegor256/cactoos/issues/1144).
